src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1144,97 +820,57 @@ private void SaveServerCredentialsToCache(Uri uri, uint authScheme, ICredentials
                 bool chunkedModeForSend = IsChunkedModeForSend(state.RequestMessage);
 
                 AddRequestHeaders(
-                    requestHandle,
+                    state.RequestHandle,
                     state.RequestMessage,
                     _cookieUsePolicy == CookieUsePolicy.UseSpecifiedCookieContainer ? _cookieContainer : null);
 
                 uint proxyAuthScheme = 0;
                 uint serverAuthScheme = 0;
-                bool retryRequest = false;
-
-                do
+                state.RetryRequest = false;
+
+                // The only way to abort pending async operations in WinHTTP is to close the WinHTTP handle.
```

Comment:
Good point. I will consider for next PR. Optimizing the cancellation patterns further is my next task.